client/extensions/woocommerce/state/sites/product-categories/test/selectors.js

```diff
@@ -67,4 +67,83 @@ describe( 'selectors', () => {
 			expect( getProductCategories( state, 345 ) ).to.equal( categories345 );
 		} );
 	} );
+
+	describe( '#getProductCategory', () => {
+		it( 'should return undefined if data is not available.', () => {
+			const state = {
+				extensions: {
+					woocommerce: {}
+				}
+			};
+
+			expect( getProductCategory( state, 1, 123 ) ).to.not.exist;
+		} );
+
+		it( 'should return undefined if data is still loading.', () => {
+			const state = {
+				extensions: {
+					woocommerce: {
+						sites: {
+							123: {
+								productCategories: LOADING,
+							}
+						}
+					}
+				}
+			};
+
+			expect( getProductCategory( state, 1, 123 ) ).to.not.exist;
+		} );
+
+		it( 'should return null if category is not found', () => {
+			const state = {
+				extensions: {
+					woocommerce: {
+						sites: {
+							123: {
+								productCategories: [
+									{ id: 1, name: 'Cat 1', slug: 'cat-1' },
+								]
+							}
+						}
+					}
+				}
+			};
+
+			expect( getProductCategory( state, 2, 123 ) ).to.equal( null );
+			expect( getProductCategory( state, { index: 0 }, 123 ) ).to.equal( null );
```

Comment:
It's to show what happens if a Category placeholder ID is passed to the selector by accident. Those are what the placeholder IDs look like for things like categories, products, variations, etc.